catkin_home/src/nav/actions/src/navServer.py

```diff
@@ -0,0 +1,133 @@
+#! /usr/bin/env python
+
+import rospy
+import actionlib
+import json
+
+import actionlib_tutorials.msg
+import actionlib_tutorial.msg
+from geometry_msgs.msg import PoseStamped
+from move_base_msgs.msg import MoveBaseActionGoal, MoveBaseGoal, MoveBaseAction
+
+class navigationServer(object):
+    # Create messages that are used to publish feedback/result
+    _feedback = actionlib_tutorial.msg.navServFeedback()
+    _result = actionlib_tutorial.msg.navServResult()
+
+    # Initialize posestamped variables
+    x=0
+    y=0
+    z=0
+    qx=0
+    qy=0
+    qz=0
+    qw=0
+
+
+    def __init__(self, name):
+        self._action_name = name
+        # Initialize Navigation Action Server
+        self._as = actionlib.SimpleActionServer(self._action_name, actionlib_tutorial.msg.navServAction, execute_cb=self.execute_cb, auto_start = False)
+        self._as.start()
+
+    def searchGoal(self, goal):
+
+       # Load data
+        i=""
+        with open('src/actionlib_tutorial/src/mock_locations.json') as x:
+            goals = json.load(x)
+
+        for pos in goals:
+            # goal.order -> the location sent by the main_engine (specify the type on nav_action/navServ.action)
+            if(pos == goal.order): 
+                i  = goals[pos]["location"]
+
+       
+        if(i == ""): return False
+
+        self.x = i["x"]
+        self.y = i["y"]
+        self.z = i["z"]
+        self.qx = i["qx"]
+        self.qy = i["qy"]
+        self.qz = i["qz"]
+        self.qw = i["qw"]
+
+        return True
+      
+    def execute_cb(self, goal):
+        # Helper variables
+        r = rospy.Rate(1)
+        success = True
+        g = rospy.Publisher('move_base_simple/goal', MoveBaseGoal, queue_size=10)
```

Comment:
Maybe in the future this `Publisher` object and the `SimpleActionClient` object could be created only once, and not in every callback, by being a member variable of the class created in the \_\_init__().